libs/MeshKernelApi/include/MeshKernelApi/Utils.hpp

```diff
@@ -397,19 +400,20 @@ namespace meshkernelapi
                             makeGridParameters.upper_right_y);
     }
 
-    template <typename ValueType>
+    template <typename InterpolatableType>
     static std::unique_ptr<meshkernel::MeshInterpolation> CreateBilinearInterpolator(const meshkernel::Mesh2D& mesh2d,
                                                                                      const GriddedSamples& griddedSamples)
     {
         meshkernel::Point origin{griddedSamples.x_origin, griddedSamples.y_origin};
         if (griddedSamples.x_coordinates == nullptr && griddedSamples.y_coordinates == nullptr)
         {
-            return std::make_unique<meshkernel::BilinearInterpolationOnGriddedSamples<ValueType>>(mesh2d,
-                                                                                                  griddedSamples.num_x,
-                                                                                                  griddedSamples.num_y,
-                                                                                                  origin,
-                                                                                                  griddedSamples.cell_size,
-                                                                                                  griddedSamples.values);
+            return std::make_unique<meshkernel::BilinearInterpolationOnGriddedSamples<InterpolatableType>>(mesh2d,
+                                                                                                           griddedSamples.num_x,
+                                                                                                           griddedSamples.num_y,
+                                                                                                           origin,
+                                                                                                           griddedSamples.cell_size,
+                                                                                                           std::span<InterpolatableType>{reinterpret_cast<InterpolatableType*>(griddedSamples.values),
+                                                                                                                                         static_cast<size_t>(griddedSamples.num_x * griddedSamples.num_y)});
```

Comment:
As we have learned recently, with null c arrays, should we check that the griddedSamples.values array is not null

Reply:
std::span has a size and empty methods. Can use one of those.